test/acceptance/user.controller.acceptance.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createClientForHandler, supertest, expect} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {ShoppingApplication} from '../..';
+
+describe('UserController', () => {
+  let app: ShoppingApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+  let userId: string;
+  const user = {
+    email: 'test@loopback.io',
+    password: 'p4ssw0rd',
+    firstname: 'Example',
+    surname: 'User',
+  };
+
+  before(givenAnApplication);
+
+  before(givenARestServer);
+
+  before(async () => {
+    await app.boot();
+    await app.start();
+  });
+
+  before(() => {
+    client = createClientForHandler(server.requestHandler);
+  });
+
+  after(async () => {
+    await app.stop();
+  });
+
+  it('creates new user when POST /users is invoked', async () => {
+    await client
+      .post('/users')
+      .send(user)
+      .expect(200)
+      // tslint:disable-next-line:no-any
+      .expect(function(res: any) {
```

Comment:
it's not all that important, but we can use the supertest's promise support here:
```ts
const response = await client.post('/users').send(user).expect(200);
// assertions
```